tests/pos/vcnull.scala

```diff
@@ -0,0 +1,26 @@
+// These issues were originally reported in SI-5866 and SI-8097
+
+case class Foo(d: Double) extends AnyVal {
```

Comment:
can you add more tests? maybe have a look on scala jira&test suite for additional ones.